File: app/src/main/java/com/example/helpwheel/ui/fuel_management/bottom_sheet_fragments/NewRideBottomSheetFragment.java
package com.example.helpwheel.ui.fuel_management.bottom_sheet_fragments;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.helpwheel.R;
import com.example.helpwheel.databinding.FragmentNewRideBottomSheetBinding;
import com.example.helpwheel.ui.fuel_management.inerface.BottomSheetCallBack;
import com.example.helpwheel.utils.Constants;
import com.example.helpwheel.utils.SharedPreferencesHolder;

import java.util.Objects;

public class NewRideBottomSheetFragment extends Fragment {
    private FragmentNewRideBottomSheetBinding binding;
    private SharedPreferences.Editor editor;
    BottomSheetCallBack callBack;
    SharedPreferencesHolder sharedPreferencesHolder;
    Constants constants;

    public NewRideBottomSheetFragment(BottomSheetCallBack callBack) {
        this.callBack = callBack;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentNewRideBottomSheetBinding.inflate(inflater, container, false);
        sharedPreferencesHolder = new SharedPreferencesHolder(requireContext());
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        SharedPreferences fuelStats = requireContext().getSharedPreferences(constants.APP_PREFERENCES, requireContext().MODE_PRIVATE);
        editor = fuelStats.edit();
        sharedPreferencesHolder.setFuelStats(fuelStats);
        sharedPreferencesHolder.setEditor(editor);
        binding.submitBtnFuel.setOnClickListener(v -> {
            String distance = Objects.requireNonNull(binding.distanceNewRideText.getText()).toString();
            String price = Objects.requireNonNull(binding.priceNewRideText.getText()).toString();
            if (!distance.isEmpty() && !price.isEmpty()) {
                editor.putFloat(constants.APP_NEW_RIDE_DISTANCE, Float.parseFloat(distance));
                editor.putFloat(constants.APP_NEW_RIDE_PRE_PRICE, Float.parseFloat(price));
                editor.apply();
                sharedPreferencesHolder.countPrice("new");
                callMethods();
                callBack.dismissBottomSheet();
            } else if (!distance.isEmpty()) {
                editor.putFloat(constants.APP_NEW_RIDE_DISTANCE, Float.parseFloat(distance));
                editor.putFloat(constants.APP_NEW_RIDE_PRE_PRICE, 0.0f);
                editor.apply();
                callMethods();
                callBack.dismissBottomSheet();
            } else if (!price.isEmpty()) {
                binding.distanceNewRideEditText.setError(getString(R.string.edit_text_odometer_error));
            } else
            callBack.dismissBottomSheet();
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        editor.apply();
    }

    private void callMethods() {
        sharedPreferencesHolder.countSpendFuel("new");
        sharedPreferencesHolder.countImpactOnEcology("new");
        sharedPreferencesHolder.calculateRemainsFuel();
    }


}

File: app/src/main/java/com/example/helpwheel/ui/trip/bottom_sheet_fragments/NewRideBottomSheetFragment.kt
package com.example.helpwheel.ui.trip.bottom_sheet_fragments

import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.helpwheel.R
import com.example.helpwheel.databinding.FragmentNewRideBottomSheetBinding
import com.example.helpwheel.ui.trip.inerface.BottomSheetCallBack
import com.example.helpwheel.ui.trip.new_ride_fragments.cost.CostNewRideViewModel
import com.example.helpwheel.ui.trip.new_ride_fragments.distance.DistanceNewRideViewModel
import com.example.helpwheel.ui.trip.new_ride_fragments.ecology.EcologyNewRideViewModel
import com.example.helpwheel.ui.trip.new_ride_fragments.remains_fuel.RemainsFuelNewRideViewModel
import com.example.helpwheel.ui.trip.new_ride_fragments.spendFuel.SpendFuelNewRideViewModel
import com.example.helpwheel.utils.APP_PREFERENCES
import com.example.helpwheel.utils.COST_NEW_RIDE
import com.example.helpwheel.utils.DISTANCE_NEW_RIDE
import com.example.helpwheel.utils.PRE_COST_NEW_RIDE

class NewRideBottomSheetFragment(private val callBack: BottomSheetCallBack) : Fragment() {
    private lateinit var binding: FragmentNewRideBottomSheetBinding
    private lateinit var fuelStats: SharedPreferences
    private lateinit var editor: SharedPreferences.Editor
    private lateinit var costNewRideViewModel: CostNewRideViewModel
    private lateinit var distanceNewRideViewModel: DistanceNewRideViewModel
    private lateinit var ecologyNewRideViewModel: EcologyNewRideViewModel
    private lateinit var remainsFuelNewRideViewModel: RemainsFuelNewRideViewModel
    private lateinit var spendFuelNewRideViewModel: SpendFuelNewRideViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentNewRideBottomSheetBinding.inflate(inflater, container, false)
        initViewModels()
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        fuelStats = requireContext().getSharedPreferences(APP_PREFERENCES, MODE_PRIVATE)
        editor = fuelStats.edit()
        binding.submitBtnFuel.setOnClickListener {
            val distance = binding.distanceNewRideText.text.toString()
            val price = binding.priceNewRideText.text.toString()
            if (distance.isNotEmpty() && price.isNotEmpty()) {
                editor.putFloat(DISTANCE_NEW_RIDE, distance.toFloat())
                editor.putFloat(PRE_COST_NEW_RIDE, price.toFloat())
                editor.apply()
                distanceNewRideViewModel.setDistanceNewRide(distance.toFloat())
                costNewRideViewModel.setCostNewRide()
                callMethods()
                callBack.dismissBottomSheet()
            } else if (distance.isNotEmpty()) {
                editor.putFloat(DISTANCE_NEW_RIDE, distance.toFloat())
                editor.putFloat(COST_NEW_RIDE, 0.0f)
                editor.apply()
                distanceNewRideViewModel.setDistanceNewRide(distance.toFloat())
                callMethods()
                callBack.dismissBottomSheet()
            } else if (price.isNotEmpty()) {
                binding.distanceNewRideEditText.error = getString(R.string.edit_text_odometer_error)
            } else
                callBack.dismissBottomSheet()
        }
    }

    private fun initViewModels() {
        costNewRideViewModel =
            ViewModelProvider(requireActivity())[CostNewRideViewModel::class.java]
        distanceNewRideViewModel =
            ViewModelProvider(requireActivity())[DistanceNewRideViewModel::class.java]
        ecologyNewRideViewModel =
            ViewModelProvider(requireActivity())[EcologyNewRideViewModel::class.java]
        remainsFuelNewRideViewModel =
            ViewModelProvider(requireActivity())[RemainsFuelNewRideViewModel::class.java]
        spendFuelNewRideViewModel =
            ViewModelProvider(requireActivity())[SpendFuelNewRideViewModel::class.java]
    }

    private fun callMethods() {
        spendFuelNewRideViewModel.setSpendFuelNewRide()
        ecologyNewRideViewModel.setCarEmissions()
        remainsFuelNewRideViewModel.setRemainsFuelNewRide()
    }


}